### ISSUE TITLE
`run_with_iobinding` is not outputting the expected result for batched input data for `T5` model running on ort `CUDA EP`. 

### ISSUE THREAD

**Issue by Ki6an** [microsoft/onnxruntime#10992]
**Describe the bug**

hey, I've implemented the code for running T5 models on `ort` with `CUDA EP`, to avoid copying data between devices I'm using `io_binding`. for binding inputs I used `io_binding.bind_input()`.  I've pre-calculated the shapes and created empty tensors (buffers) to hold the output values. for binding these outputs I used `io_binding.bind_output()`. then, I ran the Inference session with `run_with_iobinding(io_binding)`. With this setup, I was able to get `~2X` speedup for standard T5 models compared to PyTorch models on GPU (T4) for a single sequence. 

the output results were as expected for a single sequence i.e `batch_size=1`. but for `batch_size>1` I'm getting the wrong outputs. 
for eg: (just for demonstration, we use t5 for a different task)
```
batch_text_input = [
    "translate English to French: Even when you remove the bright stars, the glowing dust, and other nearby points ",
    "translate English to French: of light from the inky, dark sky, a background glow remains.",
]
```
output from onnx model running on ort (cuda ep) with io_binding 
```
["Même lorsque vous enlevez les étoiles brillantes, la poussière éclairante et d'autres points situés à proximité",
 '                                                                              ']
```
output of PyTorch models 
```
["Même lorsque vous enlevez les étoiles brillantes, la poussière éclairante et d'autres points situés à proximité", 
'd’une lumière du ciel foncé et enkyle, un éclat de fond demeure.']
```
I also ran the model with `ort.run()` method with batched input to see if there is something wrong with the exporting of the model but the results were the same as pytorch model.

as you can see from the above example the `run_with_iobinding` gives the right output only for the first index of input (batch) data and gives wrong results for the next index onwards... 

how do I fix this issue? 
any help would be appreciated, thank you.

**Urgency**
we are running a PyTorch version of the model and want to Productionalize the onnx version asap.

**System information**
- OS Platform and Distribution - `ubuntu 20.04`
- ONNX Runtime installed from (source or binary): `onnxruntime-gpu`
- ONNX Runtime version: `1.10.0`
- Python version:`3.8`
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:`11.4`
- GPU model and memory: `T4 15GB`

**Comment by wangyems** [comment]
maybe this?
https://github.com/microsoft/onnxruntime/issues/10103

**Comment by Ki6an** [comment]
@wangyems in my case I get the right output i.e (logits and past_key_values) for only the first index of the batch. but for the rest of the indexes of the batch, the logits and past_key_values are repeating...

**Comment by tianleiwu** [comment]
@Ki6an, for the same inputs, ort model shall get same output from run() and run_with_io_binding(). Could you add some debug code to dumps the inputs or outputs for them?

If you build ORT binary from source, you can add "--cmake_extra_defines onnxruntime_DEBUG_NODE_INPUTS_OUTPUTS=1" in build command line. Then set environment variables ORT_DEBUG_NODE_IO_DUMP_INPUT_DATA and ORT_DEBUG_NODE_IO_DUMP_OUTPUT_DATA to be 1. In this way, you can see inputs and outputs.

**Comment by sam-writer** [comment]
@Ki6an in another Issue, they mention having just updated the iobinding documentation... maybe that has a clue?
https://github.com/microsoft/onnxruntime/issues/10992

**Comment by Ki6an** [comment]
@tianleiwu thanks for the info, it's really helpful. 

I was able to fix this issue by binding all the inputs and output tensors as contiguous.
